feat: Add Mesh Manager RBAC page (#6113)
Kong/docs.konghq.com: app/konnect/mesh-manager/rbac.md

---

> title: Mesh Manager RBAC

Reviewer: Does this topic need to live separately? Or can it be an H2 in https://deploy-preview-6113--kongdocs.netlify.app/konnect/mesh-manager/

* **Role:** Determines what resources a user or group has access to
* **Role binding:** Determines which users or groups are assigned to a particular role

The Admin role and role binding is automatically created for you. The admin role can be used for the service mesh operators who are part of infrastructure team responsible for {{site.mesh_product_name}} deployment.

## Key mapping

### Roles

Access roles specify access levels and resources to which access is granted. Access is only defined for write operations. Read access is available to all users who have the {{site.konnect_short_name}} Mesh global control plane `viewer` role. Access roles define roles that are assigned separately to users and groups/teams using access role bindings. They are scoped globally, which means they are not bound to a mesh. 

For more information about how to configure the key mappings and RBAC settings, see [Role-Based Access Control](/mesh/latest/features/rbac/) in the {{site.mesh_product_name}} documentation.

### Role binding

| {{site.mesh_product_name}} key      | Description  |
|-----------------------------|--------------|

> | `type` | The resource type. This will be `AccessRoleBinding`. |

Reviewer: This line doesn't make sense to me. Is the type `accessrolebinding` by default? If so we can use the syntax default:`AccessRoleBinding`, but I do wonder what other types there are and if they are necessary to include.

Author: I revised it a bit, does this sound better? For role binding, it has to be `AccessRoleBinding`, but the same format is used for role mapping, so this technically could also be `AccessRole`, but then it would be for the MinK role and not the MinK role binding.

Reviewer: yes

---
title: Mesh Manager RBAC
content_type: reference
---

Mesh Manager has it's own role-based access control (RBAC) settings that are separate from the {{site.konnect_short_name}} RBAC settings. The Mesh Manager RBAC settings are specific to the meshes and mesh policies in Mesh Manager. 

To completely configure RBAC for Mesh Manager, you must configure both roles and role bindings.
* **Role:** Determines what resources a user or group has access to
* **Role binding:** Determines which users or groups are assigned to a particular role

<!-- Is the admin role included by default/automatically? -->

> ## Roles description

Reviewer 2: I think this whole section (Roles Description) can be deleted, the relevant differences for MinK are in the mappings, some comments below on that.

Reviewer 2: We can hyperlink out to the Mesh docs though.

You can use the following example roles as guides when you are configuring RBAC for Mesh Manager:

| Role                        | Description  |
|-----------------------------|--------------|
| Admin | The admin is the service mesh operator and they are a part of infrastructure team responsible for {{site.mesh_product_name}} deployment. |
| Service owner | The service owner is a part of the team that is responsible for a given service. |
| Observability operator | The observability operator is a member of the infrastructure team that is responsible for the logging, metrics, and tracing systems in the organization. Currently, those features are configured on `Mesh`, `TrafficLog`, and `TrafficTrace` objects. |
| Single mesh operator | With {{site.mesh_product_name}}, you can segment the deployment into many logical service meshes configured by the `Mesh` object. You can add a role that grants access to one specific mesh and all objects connected with this mesh instead of all meshes. |

## Key mapping

> ### Roles

Reviewer 2: I think this section can also be deleted.

Reviewer 2: Actually I think maybe we can just provide a high level description of roles, maybe stolen from the main RBAC page (and also link to more information about creating / crafting roles by linking to the main Kong Mesh RBAC page). Something like:

```
AccessRoles specify kinds of access and resources to which access is granted. Note that access is defined only for write operations. Read access is available to all users who have the Konnect Mesh Global Control Plane `viewer` role. AccessRoles define roles that are assigned separately to users and groups / teams via AccessRoleBindings. They are global-scoped, which means they are not bound to a mesh. 
```

We might want to hyperlink the Konnect Mesh Global Control Plane viewer role reference above to the page describing Konnect-wide roles that we currently have changes open for in the main MinK PR?

| {{site.mesh_product_name}} key                        | Description  |
|-----------------------------|--------------|
| `type` | The RBAC method. This will be `AccessRole`. |
| `name` | Name for the role that you want to display in the {{site.konnect_short_name}} UI. |
| `rules.types` | List of types to which access is granted. If this is empty, then access is granted to all types. Types include ? |
| `rules.names` | List of allowed names of types that access is granted to. If this is empty, then access is granted to all resources regardless of the name. |
| `rules.mesh` | Name of the mesh in {{site.konnect_short_name}} that access is granted to the resources. It can only be used with the Mesh-scoped resources. |
| `rules.access` | an action that is bound to a type.? |

> | `rules.when.sources` | a condition on sources section in connection policies (like TrafficRoute or Healtchecheck). If this value is missing, then all sources are allowed.? Is this the name of a policy/policies? |

Author: ```suggestion
| `rules.when.sources` | a condition on sources section in connection policies (like TrafficRoute or Healthcheck). If this value is missing, then all sources are allowed.? Is this the name of a policy/policies? |
```

> | `rules.when.destinations` | a condition on destinations section in connection policies (like TrafficRoute or Healtchecheck). If this value is missing, then all sources are allowed.? |

Author: ```suggestion
| `rules.when.destinations` | a condition on destinations section in connection policies (like TrafficRoute or Healthcheck). If this value is missing, then all sources are allowed.? |
```

| `rules.when.selectors` | a condition on selectors section in dataplane policies (like TrafficTrace or ProxyTemplate). |
| `rules.when.dpToken` | a condition on generate dataplane token. |

> ### Role binding

Reviewer 2: For this section, I think it only needs to be really small. I like the table you have here, added a few inline suggestions below.

| {{site.mesh_product_name}} key                        | Description  |
|-----------------------------|--------------|

> | `type` | The RBAC method. This will be `AccessRoleBinding`. |

Reviewer 2: ```suggestion
| `type` | The resource type. This will be `AccessRoleBinding`. |
```

| `name` | Name for the role that you want to display in the {{site.konnect_short_name}} UI. |

> | `subjects.type` | The type of subject you want to bind the role to. This will be `User` or `Group`. |

Reviewer 2: ```suggestion
| `subjects.type` | The type of subject you want to bind the role to. This must be one of `User` or `Group`. |
```

> | `subjects.name` | Either the email of the user or the name of the group depending on what value you used for `subjects.type`. ? |

Reviewer 2: ```suggestion
| `subjects.name` | For `subjects.type`=`User`, this should be the Konnect email address associated with them. For `subjects.type`=`Group`, this should be the name of the Konnect team you want to bind the role to. |
```

> | `subjects.name` | Either the email of the user or the name of the group depending on what value you used for `subjects.type`. | ? |

Author: ```suggestion
| `subjects.name` | Either the email of the user or the name of the group depending on what value you used for `subjects.type`. ? |
```

> | `roles` | List of roles that you want to assign to the users or groups. |

Reviewer 2: ```suggestion
| `roles` | List of roles that you want to assign to the users or groups / teams. |
```
